.github/workflows/build-base-image.yml

```diff
@@ -3,6 +3,13 @@ on:
   schedule:
     # Every Monday at 6:22am Eastern Time
     - cron: '22 10 * * 1'
+  push:
+    # Build base image when the Dockerfile or the workflow file changes
+    branches:
+      - main
+    paths:
+      - .github/workflows/build-base-image.yml
+      - docker/Dockerfile.baseimage
```

Comment:
We now update the base image on pushes to `main` that modify the base-image workflow or dockerfile. Previously this was done as part of `build.yml`, but based on the new structure, pushing this within its own workflow made more sense to me.